tripal_chado/src/Commands/ChadoCheckTermsAgainstYaml.php

```diff
@@ -235,6 +236,123 @@ protected function chadoCheckTerms_findCVProblems(&$problems, &$solutions, &$sum
+   *   Options from drush command line
+   **/
+  protected function chadoCheckTerms_findBundleProblems(&$problems, &$solutions, &$summary_rows, $options) {
+
+    // Defines which core tripal tables we want to check, and the term to use.
+    // For protocol, sep:00101 is already used, so the term is different than the bundle term
+    $migrated_tables = [
+      'analysis' => 'operation:2945',
+      'arraydesign' => 'EFO:0000269',
+      'assay' => 'OBI:0000070',
+      'biomaterial' => 'sep:00195',
+      'contact' => 'NCIT:C47954',
+      'organism' => 'OBI:0100026',
+      'project' => 'NCIT:C47885',
+      'pub' => 'TPUB:0000002',
+      'study' => 'SIO:001066',
+    ];
+
+    foreach ($migrated_tables as $table => $expected_bundle_term) {
+
+      // Lookup the term to use for records not already having a bundle term
+      $expected_term_parts = explode(':', $expected_bundle_term);
+      $expected_bundle_term_id = $this->getCVTermId($expected_term_parts[0], $expected_term_parts[1]);
+
+      // Get a list of all bundles using this table, e.g. for the analysis
+      // table we might have ['genome_annotation', 'genome_assembly', 'analysis']
+      $existing_info = [];
+      $bundles = \Drupal::entityTypeManager()
+        ->getStorage('tripal_entity_type')
+        ->getQuery()
+        ->condition('third_party_settings.tripal.chado_base_table', $table)
+        ->execute();
+      $bundle_ids = array_keys($bundles);
+      if (!$bundle_ids) {
+        continue;
+      }
+
+      // For each of these bundles, get its defining DB and
+      // accession, and then from that get its cvterm_id
+      $cvterm_id_list = [];
+      foreach ($bundle_ids as $bundle_id) {
+        $bundle = \Drupal::entityTypeManager()
+          ->getStorage('tripal_entity_type')
+          ->loadByProperties(['id' => $bundle_id]);
+        $termIdSpace = $bundle[$bundle_id]->getTermIdSpace();
+        $termAccession = $bundle[$bundle_id]->getTermAccession();
+        $cvterm_id = $this->getCVTermId($termIdSpace, $termAccession);
+        $cvterm_id_list[] = $cvterm_id;
+      }
+
+      // Get a count of records without a bundle term
+      $pkey = $table . '_id';
+      $prop_table = $table . 'prop';
+
+      // Subquery gets the list of records that already have a bundle term defined
+      $subquery = $this->chado->select('1:' . $table, 'BT');
+      $subquery->leftJoin('1:' . $prop_table, 'P', '"BT".' . $pkey . '=' . '"P".' . $pkey);
+      $subquery->condition('"P".type_id', $cvterm_id_list, 'IN');
+      $subquery->fields('BT', [$pkey]);
+
+      // Query gets the list of records without a bundle term
+      $query = $this->chado->select('1:' . $table, 'BT');
+      $query->leftJoin('1:' . $prop_table, 'P', '"BT".' . $pkey . '=' . '"P".' . $pkey);
+      $query->condition('"BT".' . $pkey, $subquery, 'NOT IN');
+      $query->addField('BT', $pkey, 'pkey_id');
+      $query->distinct();
+      $count = $query->countQuery()->execute()->fetchField();
+
```

Comment:
This makes a really big assumption which may not be correct :grimaces: It assumes that the type will be in the prop table and that is SHOULD be there. What happens if it isn't? Like for example, if this site is using Chado 1.3.3.013 and there is now a project.type_id and project-based content types that have their type stored in the base table.